controllers/userController.js

```diff
@@ -22,13 +22,83 @@ exports.loginByWechat = function(req,res){
 	util.checkParam(req.body,['code'],function(err){
 		if (err) {
 			console.log(err);
-			res.send(util.wrapBody('Internal Error','E'));
+			res.send(util.wrapBody('Invalid Parameter','E'));
 		}else{
 			login(req,res,'wechat');
 		}
 	});	
 }
 
+exports.update = function(req,res){
+	// util.checkParam(req.body,['name','nickname'],function(err){
+	// 	if (err) {
+	// 		console.log(err);
+	// 		res.send(util.wrapBody('Invalid Parameter','E'));
+	// 		return;
+	// 	}
+	// });
+
+	var id = req.token.userId;
+
+	var user = {
```

Comment:
This is not the way we talked about. eg, if nickname is not in the body, then it would still have a nickname key in user object with value undefined. I can't remember how mongo perform in this case. probably, set this key to null. That is unexpected.  Please ensure here is correct.